optimum/torch_export_patches/_patch_transformers.py

```diff
@@ -0,0 +1,1324 @@
+import inspect
+from dataclasses import dataclass
+from functools import wraps
+from typing import Any, Callable, Dict, List, Optional, Tuple
+import packaging.version as pv
+import torch
+import transformers
+from transformers.modeling_attn_mask_utils import AttentionMaskConverter
+from transformers.cache_utils import StaticCache, Cache, DynamicCache
+
+try:
+    import transformers.masking_utils
+
+    patch_masking_utils = True
+except ImportError:
+    patch_masking_utils = False
+
+
+def _has_transformers(version: str) -> bool:
+    "Returns True if transformers version is higher."
+    return pv.Version(transformers.__version__) >= pv.Version(version)
+
+def _is_torchdynamo_exporting() -> bool:
+    """Tells if :epkg:`torch` is exporting a model. Relies on ``torch.compiler.is_exporting()``."""
+
+    if not hasattr(torch.compiler, "is_exporting"):
+        # torch.compiler.is_exporting requires torch>=2.7
+        return False
+
+    try:
+        return torch.compiler.is_exporting()
+    except Exception:
+        try:
+            import torch._dynamo as dynamo
+
+            return dynamo.is_exporting()  # type: ignore
+        except Exception:
+            return False
+
+if patch_masking_utils:
```

Comment:
we already have an sdpa mask function that doesn't use vmap and supports all features of sdpa_mask, I would prefer using the registration api to register that as it seems to me less error prone than patching torch and transformers. wdyt ?